packages/dataviews/src/view-actions.js

```diff
@@ -35,84 +34,81 @@ function ViewTypeMenu( { view, onChangeView, supportedLayouts } ) {
 	}
 	const activeView = _availableViews.find( ( v ) => view.type === v.type );
 	return (
-		<DropdownSubMenu
+		<DropdownMenu
 			trigger={
-				<DropdownSubMenuTrigger
+				<DropdownMenuItem
 					suffix={
-						<>
-							{ activeView.label }
-							<Icon icon={ chevronRightSmall } />
-						</>
+						<span aria-hidden="true">{ activeView.label }</span>
 					}
 				>
-					{ __( 'Layout' ) }
-				</DropdownSubMenuTrigger>
+					<DropdownMenuItemLabel>
+						{ __( 'Layout' ) }
+					</DropdownMenuItemLabel>
+				</DropdownMenuItem>
 			}
+			placement="left-start"
 		>
 			{ _availableViews.map( ( availableView ) => {
 				return (
-					<DropdownMenuItem
+					<DropdownMenuRadioItem
 						key={ availableView.type }
-						role="menuitemradio"
-						aria-checked={ availableView.id === view.type }
-						prefix={
-							availableView.type === view.type && (
-								<Icon icon={ check } />
-							)
-						}
-						onSelect={ () => {
+						value={ availableView.type }
+						name="view-actions-available-view"
+						checked={ availableView.type === view.type }
+						onChange={ () => {
 							onChangeView( {
 								...view,
 								type: availableView.type,
 							} );
 						} }
 					>
-						{ availableView.label }
-					</DropdownMenuItem>
+						<DropdownMenuItemLabel>
+							{ availableView.label }
+						</DropdownMenuItemLabel>
+					</DropdownMenuRadioItem>
 				);
 			} ) }
-		</DropdownSubMenu>
+		</DropdownMenu>
 	);
 }
 
 const PAGE_SIZE_VALUES = [ 10, 20, 50, 100 ];
 function PageSizeMenu( { view, onChangeView } ) {
 	return (
-		<DropdownSubMenu
+		<DropdownMenu
 			trigger={
-				<DropdownSubMenuTrigger
-					suffix={
-						<>
-							{ view.perPage }
-							<Icon icon={ chevronRightSmall } />
-						</>
-					}
+				<DropdownMenuItem
+					suffix={ <span aria-hidden="true">{ view.perPage }</span> }
 				>
-					{ /* TODO: probably label per view type. */ }
-					{ __( 'Rows per page' ) }
-				</DropdownSubMenuTrigger>
+					<DropdownMenuItemLabel>
+						{ /* TODO: probably label per view type. */ }
+						{ __( 'Rows per page' ) }
+					</DropdownMenuItemLabel>
+				</DropdownMenuItem>
 			}
+			// TODO": remove?
+			placement="left-start"
 		>
 			{ PAGE_SIZE_VALUES.map( ( size ) => {
 				return (
-					<DropdownMenuItem
+					<DropdownMenuRadioItem
 						key={ size }
-						role="menuitemradio"
-						aria-checked={ view.perPage === size }
-						prefix={
-							view.perPage === size && <Icon icon={ check } />
-						}
-						onSelect={ ( event ) => {
-							// We need to handle this on DropDown component probably..
-							event.preventDefault();
+						value={ size }
+						name="view-actions-page-size"
+						checked={ view.perPage === size }
+						onChange={ () => {
 							onChangeView( { ...view, perPage: size, page: 1 } );
 						} }
 					>
-						{ size }
-					</DropdownMenuItem>
+						<DropdownMenuItemLabel>{
+							// The underlying `Truncate` element requires its children to be
```

Comment:
I'm curious why we don't do that inside the component.

Reply:
I will look into it

Reply:
Opened https://github.com/WordPress/gutenberg/pull/57261 to improve `Truncate`

Reply:
With https://github.com/WordPress/gutenberg/pull/57261 merged, I removed the workaround [`277409a` (#55625)](https://github.com/WordPress/gutenberg/pull/55625/commits/277409ae4fd600c075e57c98480889c4df88fa3d)